site/docs/main/configuration.md

```diff
@@ -1,9 +1,60 @@
 # Contour Configuration Reference
 
+- [Serve Flags](#serve-flags)
 - [Configuration File](#configuration-file)
 - [Environment Variables](#environment-variables)
 - [Bootstrap Config File](#bootstrap-config-file)
 
+## Overview
+
+There are various ways to configure Contour, flags, the configuration file, as well as environment variables.
+Contour has a precedence of configuration for contour serve, meaning anything configured in the config file is overridden by environment vars which are overridden by cli flags.
+
+## Serve Flags
+
+The `contour serve` command is the main command which is used to watch for Kubernetes resource and process them into Envoy configuration which is then streamed to any Envoy via its xDS gRPC connection.
+There are a number of flags that can be passed to this command which further configures how Contour operates. 
+Many of these flags are mirrored in the [Contour Configuration File](#configuration-file).
+
+| Flag Name         | Description        |
+|-------------------|--------------------|
+| --config-path       | Path to base configuration |
+| --incluster         | Use in cluster configuration |
+| --kubeconfig=KUBECONFIG |    Path to kubeconfig (if not in running inside a cluster) |
+| --xds-address=XDS-ADDRESS | xDS gRPC API address |
```

Comment:
The all-caps here kind of suggests that these could be environment variables. Can I suggest a format like this?
```suggestion
| `--xds-address=<ipaddr>` | xDS gRPC API address |
```
where the `<ipaddr>` could be any type like `<int>` or `<string>`.

Also note the backticks, I think that having the flags fixed-width will help them stand out more.